### ISSUE TITLE
needless nested `match`

### ISSUE THREAD

**Issue by owenthewizard** [rust-lang/rust-clippy#13773]
### What it does

Warn on a nested match that can be merged.

### Advantage

- Clarity
- Concise

### Drawbacks

- Could be less readable in non-trivial cases (not `Option`/`Result`).

### Example

```rust
match token {
    Ok(token) => match token {
        Token::A | Token::B => { ... }
    }
    Err(err) => {}
}
```

Could be written as:

```rust
match token {
    Ok(Token::A | Token::B) => { ... }
    Err(err) => {}
}
```

**Comment by y21** [comment]
This sounds like what the [`collapsible_match`](https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_match) lint looks for. That lint should probably be extended to catch this

**Comment by ibilalkayy** [comment]
Hey @owenthewizard, 

Where is this nested match present so that I can solve this problem? I have just started contributing here.

Thank you!
